CMakeLists.txt

```diff
@@ -36,6 +36,7 @@ endif()
 set(BUILD_WITH_PYTHON_3_VERSION 3.7 CACHE STRING "The version of Python 3 to build with")
 option(CMAKE_WANT_UFE_BUILD "Enable building with UFE (if found)." ON)
 option(CMAKE_WANT_MATERIALX_BUILD "Enable building with MaterialX (experimental)." OFF)
+option(CMAKE_WANT_VALIDATE_UNDO_ITEM "Enable validating undo items list." $<IF:$<STREQUAL:${CMAKE_BUILD_TYPE},Release>.OFF,ON>)
```

Comment:
Three things
1. Are you sure using a generator expression works in an option?
2. I think the period . should be a comma ,
3. In your comments you say, "It is off by default in release builds" - so doesn't that mean it should be OFF for RelWithDebInfo as well? That is a release build and the one we ship.

Reply:
Oh we ship with RelWithDebInfo? :( that is specifically the one I wanted enabled because it is the one I use locally. I also think many dev are using it and I wanted this turned on for devs by default so it catches the errors. If off by default for dev then it is much less useful.

Is there another build flag we turn on only for dev and off for release I could detect instead?

Reply:
As for the dot-vs-comma, you're right, I wonder how it worked... must fall on some valid CMake syntax by accident with the dot!